### PR TITLE
Userland: Add the rev(1) utility to reverse lines and it's manpage.

### PR DESCRIPTION
**Pull request by Mango0x45**
My first time writing C++ :)
I made sure to test it, and it worked perfectly on multiple files, piped input, and redirected files.

Also I accidentally made a typo in one of the commit messages, saying I added a manpage for `man` instead of `rev`, whoops.

### CODE REVIEW

File: Base/usr/share/man/man1/rev.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+## Synopsis
+
+```*sh
+$ rev [files...]
+```
+
+## Description
+
+`rev` reads the specified files line by line, and prints them to standard
```
Comment (dj2) [inline Base/usr/share/man/man1/rev.md:13]: Maybe something like `rev reads the specified files line by line and prints each line reversed to standard output`? When I read this the first time I thought it was going to print the lines in the right character order, but the file would be printed bottom to top.

File: Base/usr/share/man/man1/rev.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+output in reverse. If no files are specifed, then `rev` will read from standard
+input.
+
+## Examples
+
```
Comment (dj2) [inline Base/usr/share/man/man1/rev.md:18]: An example like the following would make it a bit more explicit?

```sh
$ cat foo
line1
line2
$ rev foo
1enil
2enil
```

File: Base/usr/share/man/man1/rev.md (hunk 1 above)
Comment (Mango0x45) [reply Base/usr/share/man/man1/rev.md:13]: I clarified this in my latest commit.

File: Base/usr/share/man/man1/rev.md (hunk 2 above)
Comment (Mango0x45) [reply Base/usr/share/man/man1/rev.md:18]: Good idea. I did this for both examples.

File: Base/usr/share/man/man1/rev.md (hunk 1 above)
Comment (dj2) [reply Base/usr/share/man/man1/rev.md:13]: Awesome, thank you.

**Comment by linusg** [comment]
Please observe the [contribution guidelines](https://github.com/SerenityOS/serenity/blob/master/CONTRIBUTING.md), specifically please remove the merge commit and squash any fixup commits into the appropriate previous commits.

**Comment by sin-ack** [comment]
I'm pretty sure we already have `tac`, no? Does this do something else?

**Comment by Mango0x45** [comment]
After some time googling around and figuring out how git works I managed to squash the commits. Good learning experience :P

**Pull request merged**